subsys/bluetooth/controller/hci/hci.c

```diff
@@ -1605,14 +1657,35 @@ static void le_set_scan_enable(struct net_buf *buf, struct net_buf **evt)
 	}
 
 #if CONFIG_BT_CTLR_DUP_FILTER_LEN > 0
-	/* initialize duplicate filtering */
+	/* Initialize duplicate filtering */
 	if (cmd->enable && cmd->filter_dup) {
-		dup_count = 0;
-		dup_curr = 0U;
+		if (0) {
+
+#if defined(CONFIG_BT_CTLR_SYNC_PERIODIC_ADI_SUPPORT)
+		} else if (dup_count < 0) {
```

Comment:
Any chance that `dup_count can be < 0` and not be `DUP_FILTER_DISABLED`? i.e. should/could this check be `dup_count == DUP_FILTER_DISABLED` instead?